server/src/main/java/org/cloudfoundry/identity/uaa/scim/vmidentity/VmidentityScimGroupMembershipManager.java

```diff
@@ -57,234 +59,169 @@ public VmidentityScimGroupMembershipManager(CasIdmClient casIdmClient) {
-                }
-                else
-                {
+            } else if (member.getType() == ScimGroupMember.Type.GROUP) {
+                idmPrincipal = this.idmClient.findGroup(tenant, member.getMemberId());
+                if (idmPrincipal != null) {
+                    this.idmClient.addGroupToGroup(tenant, idmPrincipal.getId(), group.getName());
+                } else {
                     logger.error("group not found  " + member.getMemberId());
                     throw new ScimResourceNotFoundException("Group not found: " + member.getMemberId());
                 }
-            }
-            else
-            {
-                throw new InvalidArgumentException(
-                    String.format("Invalid ScimMemberType '%s'", member.getType()));
+            } else {
+                throw new InvalidArgumentException(String.format("Invalid ScimMemberType '%s'", member.getType()));
             }
 
             return member;
-        }
-        catch(com.vmware.identity.idm.MemberAlreadyExistException ex)
-        {
+        } catch (com.vmware.identity.idm.MemberAlreadyExistException ex) {
             logger.debug(String.format("Scim group member already exists '%s'.", member.getMemberId()));
             throw new MemberAlreadyExistsException(
-                String.format("Scim group member already exists '%s'.", member.getMemberId()));
-        }
-        catch(InvalidPrincipalException ex)
-        {
+                    String.format("Scim group member already exists '%s'.", member.getMemberId()));
+        } catch (InvalidPrincipalException ex) {
             logger.debug(String.format("Scim resource not found '%s'.", ex.getPrincipal()));
-            throw new ScimResourceNotFoundException(
-                String.format("Scim resource not found '%s'.", ex.getPrincipal()) );
-        }
-        catch (Exception ex)
-        {
+            throw new ScimResourceNotFoundException(String.format("Scim resource not found '%s'.", ex.getPrincipal()));
+        } catch (Exception ex) {
             logger.debug(String.format("addMember '%s', '%s' failed", groupId, member.getMemberId()), ex);
-            throw new IllegalStateException(
-                String.format("addMember '%s', '%s' ", groupId, member.getMemberId()), ex);
+            throw new IllegalStateException(String.format("addMember '%s', '%s' ", groupId, member.getMemberId()), ex);
         }
     }
 
     @Override
     public List<ScimGroupMember> getMembers(String groupId, String filter, boolean includeEntities)
-            throws ScimResourceNotFoundException
-    {
-        logger.debug(
-            String.format("not supported yet getMembers '%s' '%s' '%s' ",
-               groupId, filter,
-               includeEntities ? "true" : "false" ) );
-        throw new UnsupportedOperationException();
+            throws ScimResourceNotFoundException {
+        // TODO: Figure out how to filter these...
```

Comment:
During testing we weren't able to perform proper filtering based on the 'members' field for a group. It looks like it sort of works, so I've gone ahead and re-enabled it. The updated change to implement these should be on their way in.